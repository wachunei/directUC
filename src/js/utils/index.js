export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
