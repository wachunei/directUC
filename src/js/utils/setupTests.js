// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
