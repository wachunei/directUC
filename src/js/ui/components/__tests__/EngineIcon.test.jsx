import EngineIcon from "../EngineIcon";
import { render, getElement } from "./utils";

describe("EngineIcon Component", () => {
  test("should match snapshot", () => {
    const { container } = render(<EngineIcon />);
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should match snapshot with lightTheme", () => {
    const { container } = render(<EngineIcon />, "lightTheme");
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should match snapshot with darkTheme", () => {
    const { container } = render(<EngineIcon />, "darkTheme");
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should accept props", () => {
    const { container } = render(<EngineIcon width="200px" height="200px" />);
    expect(getElement(container)).toMatchSnapshot();
  });
});
