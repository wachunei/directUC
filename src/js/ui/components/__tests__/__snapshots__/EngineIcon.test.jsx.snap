// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineIcon Component should accept props 1`] = `
<svg
  fill="none"
  height="200px"
  viewBox="0 0 16 16"
  width="200px"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M13.331 7.999c0-1.1.815-2.025 1.947-2.382a7.99 7.99 0 00-1.728-2.995c-.88.847-2.12 1.113-3.093.552-.976-.566-1.366-1.784-1.06-2.975A8.019 8.019 0 007.64 0c-.563 0-1.113.06-1.643.17.317 1.2-.07 2.433-1.056 3.003-.996.574-2.274.282-3.154-.614A7.984 7.984 0 00.001 5.617C1.134 5.974 1.947 6.9 1.947 8c0 1.098-.813 2.025-1.947 2.38a7.988 7.988 0 001.811 3.09c.88-.877 2.143-1.16 3.13-.589.973.56 1.364 1.77 1.068 2.953a7.906 7.906 0 003.374-.03c-.281-1.173.11-2.368 1.074-2.924.965-.558 2.191-.303 3.069.526a7.982 7.982 0 001.754-3.027c-1.134-.354-1.949-1.282-1.949-2.38zM7.64 11.243a3.244 3.244 0 010-6.486 3.244 3.244 0 010 6.486z"
    fill="#828282"
  />
</svg>
`;

exports[`EngineIcon Component should match snapshot 1`] = `
<svg
  fill="none"
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M13.331 7.999c0-1.1.815-2.025 1.947-2.382a7.99 7.99 0 00-1.728-2.995c-.88.847-2.12 1.113-3.093.552-.976-.566-1.366-1.784-1.06-2.975A8.019 8.019 0 007.64 0c-.563 0-1.113.06-1.643.17.317 1.2-.07 2.433-1.056 3.003-.996.574-2.274.282-3.154-.614A7.984 7.984 0 00.001 5.617C1.134 5.974 1.947 6.9 1.947 8c0 1.098-.813 2.025-1.947 2.38a7.988 7.988 0 001.811 3.09c.88-.877 2.143-1.16 3.13-.589.973.56 1.364 1.77 1.068 2.953a7.906 7.906 0 003.374-.03c-.281-1.173.11-2.368 1.074-2.924.965-.558 2.191-.303 3.069.526a7.982 7.982 0 001.754-3.027c-1.134-.354-1.949-1.282-1.949-2.38zM7.64 11.243a3.244 3.244 0 010-6.486 3.244 3.244 0 010 6.486z"
    fill="#828282"
  />
</svg>
`;

exports[`EngineIcon Component should match snapshot with darkTheme 1`] = `
<svg
  fill="none"
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M13.331 7.999c0-1.1.815-2.025 1.947-2.382a7.99 7.99 0 00-1.728-2.995c-.88.847-2.12 1.113-3.093.552-.976-.566-1.366-1.784-1.06-2.975A8.019 8.019 0 007.64 0c-.563 0-1.113.06-1.643.17.317 1.2-.07 2.433-1.056 3.003-.996.574-2.274.282-3.154-.614A7.984 7.984 0 00.001 5.617C1.134 5.974 1.947 6.9 1.947 8c0 1.098-.813 2.025-1.947 2.38a7.988 7.988 0 001.811 3.09c.88-.877 2.143-1.16 3.13-.589.973.56 1.364 1.77 1.068 2.953a7.906 7.906 0 003.374-.03c-.281-1.173.11-2.368 1.074-2.924.965-.558 2.191-.303 3.069.526a7.982 7.982 0 001.754-3.027c-1.134-.354-1.949-1.282-1.949-2.38zM7.64 11.243a3.244 3.244 0 010-6.486 3.244 3.244 0 010 6.486z"
    fill="#828282"
  />
</svg>
`;

exports[`EngineIcon Component should match snapshot with lightTheme 1`] = `
<svg
  fill="none"
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M13.331 7.999c0-1.1.815-2.025 1.947-2.382a7.99 7.99 0 00-1.728-2.995c-.88.847-2.12 1.113-3.093.552-.976-.566-1.366-1.784-1.06-2.975A8.019 8.019 0 007.64 0c-.563 0-1.113.06-1.643.17.317 1.2-.07 2.433-1.056 3.003-.996.574-2.274.282-3.154-.614A7.984 7.984 0 00.001 5.617C1.134 5.974 1.947 6.9 1.947 8c0 1.098-.813 2.025-1.947 2.38a7.988 7.988 0 001.811 3.09c.88-.877 2.143-1.16 3.13-.589.973.56 1.364 1.77 1.068 2.953a7.906 7.906 0 003.374-.03c-.281-1.173.11-2.368 1.074-2.924.965-.558 2.191-.303 3.069.526a7.982 7.982 0 001.754-3.027c-1.134-.354-1.949-1.282-1.949-2.38zM7.64 11.243a3.244 3.244 0 010-6.486 3.244 3.244 0 010 6.486z"
    fill="#828282"
  />
</svg>
`;
